src/torrent.cpp

```diff
@@ -2328,7 +2328,16 @@ namespace libtorrent
 		if (req.downloaded < 0) req.downloaded = 0;
 
 		req.event = e;
-		error_code ec;
+		if (!m_ses.m_settings.anonymous_mode)
+		{
+			tcp::endpoint ep;
+#if TORRENT_USE_IPV6
+			ep = m_ses.get_ipv6_interface();
+			if (ep != tcp::endpoint()) req.ipv6 = ep.address();
```

Comment:
GCC complains about that `address` type can not be directly converted to `address_v6` nor `address_v4`.